setup/Core/Utilities/PathUtils.cs

```diff
@@ -0,0 +1,21 @@
+namespace Terraria.ModLoader.Setup.Core.Utilities
+{
+	public static class PathUtils
+	{
+		public static string GetCrossPlatformFullPath(string path)
```

Comment:
.NET does not correctly resolve the Home directory shortcut `~` by itself. So I had to manually write code to account for this. Otherwise paths input by the user might not work although the path exists. If you use `Path.GetFullPath("~")` you get the current working directory combined with `~` at the end.